src/script/view_model/ContentViewModel.ts

```diff
@@ -158,84 +261,35 @@ export class ContentViewModel {
       openNotificationSettings = false,
     } = options;
 
-    const {rightSidebar} = useAppMainState.getState();
-    const {contentState, setContentState} = useAppState.getState();
-
     if (!conversation) {
-      rightSidebar.close();
-      return this.switchContent(ContentState.CONNECTION_REQUESTS);
+      return this.handleMissingConversation();
     }
 
     try {
-      const conversationEntity = isConversationEntity(conversation)
-        ? conversation
-        : await this.conversationRepository.getConversationById({domain: domain || '', id: conversation});
+      const conversationEntity = await this.getConversationEntity(conversation, domain);
 
       if (!conversationEntity) {
-        rightSidebar.close();
-
-        throw new ConversationError(
-          ConversationError.TYPE.CONVERSATION_NOT_FOUND,
-          ConversationError.MESSAGE.CONVERSATION_NOT_FOUND,
-        );
+        return this.closeRightSidebarAndThrowError();
```

Comment:
Not sure I understand why extracting this code to a function if it's not used anywhere else. 
The previous version was more explicit I feel

Reply:
But here it does 2 actions, (closing **and** throwing). The rule of thumb is that naming a function with **and** is usually a sign that the single responsibility principle is broken. 

There are 2 types of function: 
- glue functions (that are called from the UI and that could perform a lot of actions in a single call)
- utility functions (that are focused on doing a single task). 

In this case `showConversation` is a glue function (that will perform a lot of tasks to get the conversation ready to be shown).

Reply:
got it, resolved it.